### ISSUE TITLE
Things I don't understand

### ISSUE THREAD

**Issue by anufrijev** [valango/simplico#2]
Hi everybody, Ilja here. As I'm spectacularly out of my depth in this project I hope I can get some things cleared up in here. Hopefully this can be useful to more than just me!

Thanks to Margus' email I got the necessary things installed on my desktop Windows 7! I don't have the mac I had on the meeting anymore (wasn't mine) so I'll need to work on Windows platform.

Then I tried to run (and understand) the commands in Villem's email:

**sh bin/glp** didn't work originally, said gulp wasn't found. Googled, discovered "npm install gulp -g" fixed the problem. I'm not sure what gulp does. Found [gulp readme](https://github.com/gulpjs/gulp/blob/master/README.md), guess it automates things so I don't need to worry about them. Should I know more about gulp presently?
![image](https://cloud.githubusercontent.com/assets/13501158/8939224/d0e72018-356b-11e5-87c1-82c3875912f9.png)
Do not know yet what lint, watch, hint-watch and default are. Feel pretty confused about glp in general.

**sh bin/srvdev**  starts the server, right? So we can see the map on localhost. That worked fine for me. I read about *nodemon* but I'm not sure how exactly it ties into the server idea. Does it restart server "live", whenever a change is made?

**sh bin/srvdbg** [is a command that logs that server debug](http://expressjs.com/guide/debugging.html)? Is this something that finds and logs bugs about server?

**sh bin/tester** wanted karma. I tried [everything here] (https://karma-runner.github.io/0.12/intro/installation.html), I think after "npm install -g karma-cli" it started working.
![image](https://cloud.githubusercontent.com/assets/13501158/8939403/f1b1dd28-356c-11e5-8a09-254ac4e87c06.png)
This tests the browser for compatibility? I read about karma [here](https://www.npmjs.com/package/karma).

As I am a front-end guy, I tried to figure out eventist like the email said, too. "npm install eventist" produced this:
![image](https://cloud.githubusercontent.com/assets/13501158/8939519/93eb9836-356d-11e5-940f-950b64020459.png)
but I think [(found here)](https://www.npmjs.com/package/eventist)
"bower install eventist" worked. I'm not sure what exactly eventist does.
As for readme, I found some syntax there challenging for now making it hard to "play" with the examples. Perhaps someone could help out.

In conclusion, I got very hungry while figuring out (or not) and writing this post. Gotta eat something. :)

**Comment by valango** [comment]
Here I try to provide few quick answers.

**Gulp** is a tools runner increasingly popular in Node.js community. In our project. it runs config/Gulpfile.js - take a look at this.
If invoked without task name (like "lint"), it will run "default" task. Now, go to GitHub global search or npmjs.org and enter the name of any module you see require()'d in the start of the Gulpfile - by doing so you will go to the components homepage/repository and see what it does. Right now, if you create a new javascript file or modify the existing one, glp will check your coding style and syntax.

**srvdbg** runs the server with verbose log output - you will see on terminal window what requests it receives and what it does.

** bin/ ** contains command scripts in text format - examine the contents.

** karma ** for more info, check the http://karma-runner.github.io - there is also bunch of youtube videos available - these may gibe you a smoother start. But at least for the beginning I'll take care of karma configuration and writings tests - we'll come back to this together a little bit later.

**eventist** install problem seems to boil down to examples/init.sh script. Examine it and try to execute the commands manually from the command line. Or send me resulting npm-debug.log file as advised.

I hope it helps.

**Comment by anufrijev** [comment]
Thanks, Villem!
About eventist, bower install eventist does this:
![image](https://cloud.githubusercontent.com/assets/13501158/8954768/8a16ebe8-35f0-11e5-965c-ac15c70c3f2f.png)
Is there still something wrong with it?

Command line doesn't seem to react when I try to feed it lines like
if [ ! -d examples/node/node_modules ]; then
  mkdir examples/node/node_modules
fi

and I'm not sure how else to execute it manually.
I've also sent you the error log.

**Comment by valango** [comment]
The `init.sh` is bash (an unix command line shell) script. You can google on "bash or sh under windows" or something. But as a quick fix, here comes human tongue translation.

**NB:** Windows uses `'\'` as a file path delimiter, while all the civilized world uses `'/'` - please correct accordingly!

1. create directory `examples/node/node_modules/` ;
2. create directory `examples/node/node_modules/eventist/` ;
3. copy the `package.json` file into it;
4. create directory `examples/node/node_modules/eventist/lib/` ;
5. copy the files from `lib/` directory into it.

All this can be done using the windows standard file explorer or how they call this graphical interface everybody uses. :)
I hope it helps - just let me know!

**Comment by anufrijev** [comment]
Where do I create the directory examples/node/node_modules/?
in C:\Program Files\nodejs\node_modules\npm?

**Comment by valango** [comment]
No! Ol the stuff above holds when you are already sitting in the eventist project directory that was created by `npm install eventist` command - or didn't it create any directory at all?

**Comment by anufrijev** [comment]
Here is the full list of my search result for directories. Which one is the one to work with?
There's one with simplico/components and one with bower.
![image](https://cloud.githubusercontent.com/assets/13501158/8959069/64ca8e28-3611-11e5-8eec-1a83598cd2c2.png)

**Comment by MrLehiste** [comment]
I can't seem to be getting it to work either:

cd ~/simplico/vendor/eventist/examples/node
node console.js


$ node console.js
module.js:338
    throw err;
          ^
Error: Cannot find module 'eventist'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (c:\Users\MLPalm\simplico\vendor\eventist\examples\nod
e\console.js:13:11)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)

**Comment by valango** [comment]
Re-defined as issue #5
